compiler/luci/pass/src/SubstitutePadV2ToPadPass.cpp

```diff
@@ -0,0 +1,358 @@
+      v2.at(to) = v.at(from);
+    }
+
+    v = v2;
+  }
+};
+
+struct ReshapingNode
+{
+  /// @brief Check if node is ReshapingNode
+  static bool check(loco::Node *node)
+  {
+    if (dynamic_cast<luci::CircleTranspose *>(node))
+      return true;
+    //
+    // add more 'reshaping op'
+    //
+    return false;
+  }
+
+  /// @brief Retuen node's input
+  static loco::Node *input(loco::Node *node)
+  {
+    if (auto transpose = dynamic_cast<luci::CircleTranspose *>(node))
+      return transpose->a();
+    //
+    // add more 'reshaping op'
+    //
+    throw std::runtime_error("Not yet supported reshaping op");
+  }
+};
+
+/// @brief Get the only successor node of type LocoNodeT
+template <typename LocoNodeT> LocoNodeT *get_only_succ(loco::Node *parent)
+{
+  assert(parent);
+
+  auto successors = loco::succs(parent);
+  if (successors.size() != 1)
+    return nullptr;
+
+  return dynamic_cast<LocoNodeT *>(*successors.begin());
+}
+
+/// @brief Get only successor node
+loco::Node *get_only_succ(loco::Node *parent)
+{
+  assert(parent);
+
+  auto successors = loco::succs(parent);
+  if (successors.size() != 1)
+    return nullptr;
+
+  return *successors.begin();
+}
+
+// Check condition C1)
+bool positive_or_zero(loco::Node *ifm)
+{
+  assert(ifm);
+
+  if (ReshapingNode::check(ifm))
+    return positive_or_zero(ReshapingNode::input(ifm));
+
+  // Since Relu.output[i] >= 0
+  if (dynamic_cast<luci::CircleRelu *>(ifm))
+    return true;
+  if (auto conv = dynamic_cast<luci::CircleConv2D *>(ifm))
+  {
+    if (conv->fusedActivationFunction() == luci::FusedActFunc::RELU)
+      return true;
+    // Add more FusedActFunc
+  }
+  // Add more ops of which output[i] >= 0
+
+  return false;
+}
+
+template <loco::DataType DT> bool has_all_positive_values(luci::CircleConst *node)
+{
+  // Only numeric datatype is allowed
+  static_assert(DT != loco::DataType::Unknown);
+  static_assert(DT != loco::DataType::STRING);
+
+  assert(node);
+
+  auto size = node->size<DT>();
+  for (decltype(size) t = 0; t < size; t++)
+  {
+    typename loco::DataTypeImpl<DT>::Type val = node->at<DT>(t);
+    if (val <= 0)
+      return false;
+  }
+
+  return true;
+}
+
+// To check condition C2)
+bool has_all_positive_values(luci::CircleConst *node)
+{
+  assert(node);
+
+  // Must be same type as tensor (https://www.tensorflow.org/api_docs/python/tf/pad)
+  if (node->dtype() == loco::DataType::FLOAT32)
+    return has_all_positive_values<loco::DataType::FLOAT32>(node);
+  // Add more datatype
+
+  throw std::runtime_error("Not yet supported datatype");
+}
+
+bool used_by_maxpool_only(luci::CircleNode *node, Paddings &paddings)
+{
+  auto successor = get_only_succ(node);
+  if (auto maxpool = dynamic_cast<luci::CircleMaxPool2D *>(successor))
+  {
+    // Let's check condition C4)
+    if (paddings.smaller_than(maxpool->filter()->h(), maxpool->filter()->w()))
+      return true;
+  }
+
+  if (auto transpose = dynamic_cast<luci::CircleTranspose *>(successor))
```

Comment:
Below `BEFORE` - `AFTER` diagram didnot show about `Transpose`
I think it would be better to show in the diagram as case 2

Reply:
OK. I will add. Thanks!